### ISSUE TITLE
Allow constructing a notification handle from an existing notification's ID

### ISSUE THREAD

**Issue by Forkk** [hoodie/notify-rust#17]
It would be useful to be able to query the ID of a notification and then construct a `NotificationHandle` later from the ID. This would make it possible to store the ID in a file and access the notification later, possibly from another process.

An example use case for something like this can be found in [notify-send.sh](https://github.com/vlevit/notify-send.sh). Currently, I can't replicate this, as `notify-rust` doesn't expose notifications' IDs.

**Comment by hoodie** [comment]
Hi,
it used to be possible to just update notifications via the ID that was returned from the DBUS send method, I could certainly expose the integer id again and offer some interface to do that. Unfortunately that would only be have way successful, as @panicbit noted and fixed in this [PR](https://github.com/hoodie/notify-rust/pull/5).
**TL;DR:** some server implementations are bit weird and require the dbus connection be kept alive, which is what `NotificationHandle` does. So I can surely expose the old way of doing this again, but it would fail for some servers.

Another idea, if you really want to transport a notification across processes is that you could use something like an ipc-channel. Can you tell me about your setup? NotificationDaemon and usecase?

**Comment by panicbit** [comment]
I think the problem was that a notification-daemon couldn't know anymore whom to send interaction events back. But in case you only want to update a notification, an id should suffice.
I propose we add something like `fn into_id(self) -> u32` to `NotificationHandle`, though, I'm not sure about wether we should consume the handle or not.

**Comment by hoodie** [comment]
You could also just make the id field readable, or add a getter `NotificationHandle::id()`.

I was thinking about when you would want to have really long lived notifications like this, that could be used from different places. Implementations like plasma's will reuse a visual notification anyway, if the appname is the same. But it would be awesome if desktops like kde or gnome or etc could integrate a notification center like windows or MacOS or even Android and IOS do. But for that we ought to propose changes to the standard :D

**Comment by hoodie** [comment]
*mind the updated previous post*
I would prefer something along the lines of `NotificationHandle::from_id(u32)`, but it holds this connection, I would have to look wether this could be made an `Option<Connection>` without any api breakage.

**Comment by Forkk** [comment]
My use case for this is replicating the `notify-send.sh` script I linked in the initial issue. The script has a flag that causes it to print out the notification's ID, and another flag that replaces a notification with a given ID. Since this requires exposing the ID, I can't just use IPC.

**Comment by hoodie** [comment]
turns out it wasn't that much of a deal: https://github.com/hoodie/notify-rust/blob/master/examples/update.rs
